zio-http/src/main/scala/zhttp/http/Http.scala

```diff
@@ -421,6 +422,24 @@ sealed trait Http[-R, +E, -A, +B] extends (A => ZIO[R, Option[E], B]) { self =>
       case RunMiddleware(app, mid) => mid(app).execute(a)
 
       case When(f, other) => if (f(a)) other.execute(a) else HExit.empty
+
+      case Combine(self, other) => {
+        self.execute(a) match {
+          case HExit.Effect(zio) => {
+            Effect(
+              zio.foldM(
+                {
+                  case Some(error) => ZIO.fail(Option(error.asInstanceOf[E]))
+                  case None        => other.execute(a).toZIO
+                },
+                b => ZIO.succeed(b.asInstanceOf[B]),
+              ),
+            )
+          }
```

Comment:
Why can't we do something like 

```scala
self.execute(a) match {
  case HExit.Empty => other.execute(a)
  case hExit => hExit
}
```

Reply:
Currently, in the case of None, the other app executes
IF we return the effect as it is then we won't be handling the failure correctly? And In no case, the other app would execute.